src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -259,8 +259,8 @@ def __call__(
             timesteps_tensor = torch.tensor(self.scheduler.timesteps.copy(), device=self.device)
 
         # if we use LMSDiscreteScheduler, let's make sure latents are multiplied by sigmas
-        if isinstance(self.scheduler, LMSDiscreteScheduler):
-            latents = latents * self.scheduler.sigmas[0]
+        if self.scheduler.type == SchedulerType.CONTINUOUS:
+            latents = latents * self.scheduler.init_sigma
```

Comment:
A step in the right direction. I'd love to get rid of the `if` entirely, but as long as we have it, defining a scheduler.type enum is much preferable to isinstance!

Reply:
@anton-l can also be convinced here to change it to something that might be cleaner (again if we don't have to force a function upon DDIM or DDPM)

Overall, I feel quite strongly about the following though:
- Let's not make easy schedulers more complex because we'd like to support newer, more complex schedulers
- Forcing every scheduler to implement a certain method that can grow arbitrary in complexity is much worse that **educating** if statements with a nice comment.